spec/lib/bip70/payment_request_spec.rb

```diff
@@ -0,0 +1,131 @@
+require 'spec_helper'
+
+RSpec.describe StraightServer::Bip70::PaymentRequest do
+
+  let(:gateway) { StraightServer::GatewayOnConfig.find_by_id(1) }
+  let(:order) { create(:order, gateway_id: gateway.id) }
+
+  it 'create payment request without CA certificates' do
+    serialized_payment_request = StraightServer::Bip70::PaymentRequest.new(order: order).to_s
+    payment_request = Payments::PaymentRequest.parse(serialized_payment_request)
+
+    expect(payment_request.payment_details_version).to eq(1)
+    expect(payment_request.pki_type).to eq('none')
+    expect(payment_request.pki_data).to eq('')
+
+    serialized_payment_details = payment_request.serialized_payment_details
+    payment_details = Payments::PaymentDetails.parse(serialized_payment_details)
+
+    expect(payment_details.network).to eq('test')
+    expect(payment_details.memo).to eq('Payment request for GearPoweredMerchant')
+    expect(payment_details.payment_url).to eq('')
+    expect(payment_details.merchant_data).to eq('')
+    expect(payment_details.expires - payment_details.time).to eq(gateway.orders_expiration_period)
+    expect(payment_details.outputs.size).to eq(1)
+
+    output = payment_details.outputs.first
+    expect(output.amount).to eq(order.amount)
+    expect(output.script).to eq(order.address)
+  end
+
+  it 'create payment request for mainnet' do
+    gateway = StraightServer::GatewayOnConfig.find_by_id(2)
+    order = create(:order, gateway_id: gateway.id)
+
+    serialized_payment_request = StraightServer::Bip70::PaymentRequest.new(order: order).to_s
+    payment_request = Payments::PaymentRequest.parse(serialized_payment_request)
+    serialized_payment_details = payment_request.serialized_payment_details
+    payment_details = Payments::PaymentDetails.parse(serialized_payment_details)
+
+    expect(payment_details.network).to eq('main')
+  end
+
+  it 'create payment request with CA certificates' do
+    certificates =
+      "-----BEGIN CERTIFICATE-----\n" +
```

Comment:
I think, it's better to read them from the .crt/.key files, just like in the production env.